components/logging/org.wso2.carbon.logging.appender.http/src/main/java/org/wso2/carbon/logging/appender/http/SecuredHttpAppender.java

```diff
@@ -356,21 +399,80 @@ private String resolveSecretPassword(String password) {
         return password;
     }
 
+    private void printWarningLogOnRemoteServerFailure() {
+
+        long diskUsage = persistentQueue.getCurrentDiskUsage();
+        switch (this.failureWarningLevel) {
+            case NONE:
+                getStatusLogger().warn("Remote log publishing failure : Unable to publish logs to the remote server. " +
+                        "Please check the remote server status.");
+                this.failureWarningLevel = FailureWaringLevel.INITIAL;
+                break;
+            case INITIAL:
+                if (diskUsage > persistentQueue.getMaxDiskSpaceInBytes() / 2) {
+                    getStatusLogger().warn("Remote log publishing failure : The number of logs in the queue has exceeded" +
+                            " 50% of the allocated queue limit. Please check the remote server status.");
+                    this.failureWarningLevel = FailureWaringLevel.HALF_QUEUE_SIZE;
+                }
+                break;
+            case HALF_QUEUE_SIZE:
+                if (diskUsage > persistentQueue.getMaxDiskSpaceInBytes() * 0.9) {
+                    getStatusLogger().warn("Remote log publishing failure : The number of logs in the queue has exceeded" +
```

Comment:
check usage from disk here

Reply:
Disk usage check was implemented inside the new block creation and deletion functions since those are the places where the disk usage changes are happening.  Then the value accessed here is accurate since the value kept on memory is always accurate.